controller_manager/include/controller_manager/controller_loader.h

```diff
@@ -74,7 +74,7 @@ class ControllerLoader : public ControllerLoaderInterface
 private:
   std::string package_;
   std::string base_class_;
-  boost::shared_ptr<pluginlib::ClassLoader<T> > controller_loader_;
+  std::shared_ptr<pluginlib::ClassLoader<T> > controller_loader_;
```

Comment:
`pluginlib::UniquePtr<T>`?

Reply:
Good point! Adding...

Reply:
Hmm... I've changed my mind on this one. That typedef is only used as return type of `ClassLoader::createUniqueInstance()` while in this case it's merely a local variable of `ClassLoader<T>`

Reply:
My rationale was that we don't need shared ownership in this case.
In general `unique_ptr`  shoud be preferred over `shared_ptr`.

Reply:
That's true. How about changing it to `std::unique_ptr` here?

Reply:
I would stick to `UniquePtr`, because this is used in the `pluginlib` code.

Reply:
You're right, `std::unique_ptr` is the way to go!